### DISCUSSION TITLE
Add ability to display bundle component prices and discounts

### DISCUSSION THREAD

**Discussion by a47ae** [Shopify/function-examples#444] [Feedback: Cart Transform API]
Enhancing the visibility of individual component pricing and discounts within bundle offers would significantly improve user transparency and decision-making.

In our current setup, the cart's `expand operation` allows adding products with additional fees, but users are only presented with the aggregate price.

If we take the example from the official documentation, in a bundle like "Awesome TV with Warranty" sold for $1150.00, users lack the ability to discern the separate costs of the TV and the warranty.

This lack of detailed pricing information may lead to confusion among users as they don't know what exactly they are paying for.

**Comment by phpjscoding** [comment]
you can use the Liquid property "item.item_components " in the cart to loop through item and display title and price 
`  <ul>  
             {% for comp in item.item_components %}
             <li>  {{ comp.title | escape }}--{{ comp.price | money }}</li>
            {% endfor %}
 </ul>`

**Reply by a47ae** [reply]
Hey, thank you for your answer, but to clarify, I am talking about the Checkout and not the cart page. :)

**Comment by Kaniwong** [comment]
![image](https://github.com/Shopify/function-examples/assets/46040581/dfd0726d-a292-4398-bdd2-9ed2532feab8)
I have the same needs, customers should know how much they just saving. Is there any updates I can use for it?

**Comment by dnpg** [comment]
Back to the original request, it would be great to have the option to display the expanded items prices if required, and have the flexibility to choose which ones to show, in case it's needed:

```
...
"expandedCartItems": [
        {
          "merchandiseId": "gid://shopify/ProductVariant/1",
          "quantity": 1,
          "displayPrice": true
          "price": {
            "adjustment": {
              "fixedPricePerUnit": {
                "amount": "1000.00"
              }
            }
          }
        },
        {
          "merchandiseId": "gid://shopify/ProductVariant/2",
          "quantity": 1,
          "displayPrice": false,
          "price": {
            "adjustment": {
              "fixedPricePerUnit": {
                "amount": "150.00"
              }
            }
          }
        }
      }
```

<img width="539" alt="Screenshot 2024-07-22 at 6 20 02 PM" src="https://github.com/user-attachments/assets/3f53d484-0fc6-4179-b7fa-ff4aba18aca4">
